src/app/Main/Body/QueryPage/QueryPage-Container.component.tsx

```diff
@@ -570,12 +523,73 @@ export function QueryPageContainerComponent(): JSX.Element {
                 countedIsolatesResponse.data;
             const nrOfSelectedIsolates = isolateCountProp.totalNumberOfIsolates;
             const isolateCountGroups = isolateCountProp.groups;
-            setDataContext(isolateCountGroups, nrOfSelectedIsolates);
+            setDataContext(isolateCountGroups, nrOfSelectedIsolates, uniqueValues);
             setNrOfSelectedIsol(nrOfSelectedIsolates);
         }
         setDataIsLoading(false);
     };
 
+    const fetchNewFilterValues = async (): Promise<void> => {
+        setConditionalValuesAreLoading(true);
+        const newUniqueValues = _.cloneDeep(uniqueDataValues);
+        const filterStatusList: number[] = [];
+        const newFilterFor: string[] = [...filter.displayedFilters]

```

Comment:
It is also necessary to get the conditional filter values for the selected values in "Row" and "Column" so that the values are displayed correctly in the table. (Not only for the displayed filters) This is the main part of solving the bug.

Reply:
Thx, i should have read the source ticket.
In general it is not a good idea to mix responsibilities here. The table data should be generated independently from the values for the filter selectors. So for later refactoring of the table container, it should call the api independently and ask for the conditional values itself. Another possibility would be that the tables just filters the data to display for empty rows/cols without even knowing about conditional filters.
So your app logic has become so complex that you should think about clearly separating responsibilities (so not doing url parsing, api fetching, data mapping and updating state all in one service, but clearly separating those concerns.) Services could look like this:
urlParsingService = read url and map it to an object containing the different initial settings
urlSettingService = update the url with new settings
fetchServices = call api and map the data to an appropriate format (Mappers could be also a separate services)
dataCreationServices = create data for the sub components
Containers = read data from state and context and call the services and update state and context
Each of those services should be responsible for one peace of data (filterData, tableData and so on, e.g. service should contain only one api call)

PS: Can you tell me if you got a notification of this comment? Request is closed, so i wonder if comments are ignored.